src/Reflex/Workflow.hs

```diff
@@ -9,37 +9,42 @@
 --   "workflows."
 module Reflex.Workflow (
     Workflow (..)
+  , runWorkflow
   , workflow
   , workflowView
   , mapWorkflow
   , mapWorkflowCheap
   ) where
 
 import Control.Arrow ((***))
+import Control.Monad ((<=<))
 import Control.Monad.Fix (MonadFix)
 
 import Reflex.Class
 import Reflex.Adjustable.Class
-import Reflex.Network
-import Reflex.NotReady.Class
 import Reflex.PostBuild.Class
 
 -- | A widget in a workflow
 -- When the 'Event' returned by a 'Workflow' fires, the current 'Workflow' is replaced by the one inside the firing 'Event'. A series of 'Workflow's must share the same return type.
 newtype Workflow t m a = Workflow { unWorkflow :: m (a, Event t (Workflow t m a)) }
 
--- | Runs a 'Workflow' and returns the 'Dynamic' result of the 'Workflow' (i.e., a 'Dynamic' of the value produced by the current 'Workflow' node, and whose update 'Event' fires whenever one 'Workflow' is replaced by another).
-workflow :: forall t m a. (Reflex t, Adjustable t m, MonadFix m, MonadHold t m) => Workflow t m a -> m (Dynamic t a)
-workflow w0 = do
-  rec eResult <- networkHold (unWorkflow w0) $ fmap unWorkflow $ switch $ snd <$> current eResult
-  return $ fmap fst eResult
-
--- | Similar to 'workflow', but outputs an 'Event' that fires whenever the current 'Workflow' is replaced by the next 'Workflow'.
-workflowView :: forall t m a. (Reflex t, NotReady t m, Adjustable t m, MonadFix m, MonadHold t m, PostBuild t m) => Workflow t m a -> m (Event t a)
-workflowView w0 = do
-  rec eResult <- networkView . fmap unWorkflow =<< holdDyn w0 eReplace
```

Comment:
In `a' <- networkView =<< holdDyn a0 a`, is it safe to assume the post-build occurence of `a'` must always be `a0` ? If not, this PR needs to be reworked